### ISSUE TITLE
Basic authentication was removed but kubectl config set-credentials can set basic auth infos

### ISSUE THREAD

**Issue by zwtxy** [kubernetes/kubectl#1389]

**Comment by charles-chenzz** [comment]
I run ``kubectl config set-credentials cluster-admin --username=admin --password=whatwillhappenhere`` which is set basic authn, and kubectl didn't warn me. and I run ``kubectl config view`` and it seems like set the user and password in.

<img width="781" alt="image" src="https://user-images.githubusercontent.com/73322208/225626374-3c74f716-c41d-4251-8e8b-651bf0bc5a00.png">
<img width="517" alt="image" src="https://user-images.githubusercontent.com/73322208/225626509-9fa082f3-a978-449a-885e-0411323c29ed.png">

**Comment by gxwilkerson33** [comment]
I have a branch with the password option removed. what are the requirements here? Do we need to warn when user tries to set basic auth? I also need direction to make sure this does not cause auth problems other places? Thanks

**Comment by mpuckett159** [comment]
/triage accepted
We will begin a deprecation process for this and add a warning for this.

**Comment by gxwilkerson33** [comment]
@mpuckett159 
how complicated is the deprecation process? do you think i can take this issue? thanks

**Comment by mpuckett159** [comment]
I'm not 100% on the process but I believe we need to add a deprecation warning for 1 (2?) release cycles before we can actually remove the flag.

**Comment by brianpursley** [comment]
Deprecation policy for CLI is 12 months (or 2 releases, whichever is longer):
https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli

**Comment by gxwilkerson33** [comment]
should username be deprecated as well as password mentioned in the issue. Since auth will need to be done with token. username will cause an error or be set to an empty string in the current implementation.

**Comment by brianpursley** [comment]
> Should username be deprecated as well as password mentioned in the issue. Since auth will need to be done with token. username will cause an error or be set to an empty string in the current implementation.

Yeah.  Double check, but I'm pretty sure username is only used for basic auth, so it should be deprecated too.

**Comment by ibihim** [comment]
/remove-lifecycle rotten

How to do it: https://github.com/kubernetes/kubernetes/pull/117874#issuecomment-1540914797
